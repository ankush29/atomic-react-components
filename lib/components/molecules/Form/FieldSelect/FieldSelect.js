// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import Label from '../../../atoms/Label';
import Select from '../../../atoms/Select';
import FieldError from '../../../atoms/FieldError';
import styles from './FieldSelect.style';

type Props = {
  id: string,
  className: string,
  name: string,
  label: Node,
  labelProps?: Object,
  fieldProps?: Object,
  onChange?: Function,
};

const FieldSelect = ({
  id,
  className,
  name,
  label,
  onChange,
  labelProps,
  fieldProps,
  ...others
}: Props): Node => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>
      {label}
    </Label>
    <Field name={name} {...fieldProps}>
      {({ field, form }) => {
        const { handleChange } = form;
        return (
          <Select
            {...others}
            {...field}
            onChange={e => {
              handleChange(e);
              if (typeof onChange === 'function') onChange(e, form);
            }}
          />
        );
      }}
    </Field>
    <ErrorMessage name={name} component={FieldError} />
  </div>
);

FieldSelect.defaultProps = {
  labelProps: {},
  fieldProps: {},
  onChange: null,
};

export default styled(FieldSelect)`
  ${styles};
`;
export { FieldSelect as FieldSelectVanilla };
