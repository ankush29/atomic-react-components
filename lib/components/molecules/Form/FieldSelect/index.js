// @flow

export { default } from './FieldSelect';
