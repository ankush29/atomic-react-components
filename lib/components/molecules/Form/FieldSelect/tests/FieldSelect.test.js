import React from 'react';
import { shallowWithTheme, renderWithTheme } from '../../../../../../__mock__/themeMock';
import { FieldSelectVanilla } from '../FieldSelect';

describe('FieldInput Component', () => {
  let FieldInput = '';

  test.only('should render correctly', () => {
    FieldInput = shallowWithTheme(<FieldSelectVanilla />);
    expect(FieldInput).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Select Fav Color',
      options: ['red', 'black', 'blue'],
    };
    FieldInput = renderWithTheme(<FieldSelectVanilla {...props} />);
    expect(FieldInput.find('label').text()).toEqual('Select Fav Color');
  });

  test('should render select tag', () => {
    const props = {
      options: ['red', 'black', 'blue'],
    };
    FieldInput = renderWithTheme(<FieldSelectVanilla {...props} />);
    expect(FieldInput.find('select')).toHaveLength(1);
  });
});
