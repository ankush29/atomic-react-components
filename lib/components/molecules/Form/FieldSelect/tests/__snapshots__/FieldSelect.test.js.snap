// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <FieldSelect
    errors={Object {}}
    fieldProps={Object {}}
    labelProps={Object {}}
    onChange={null}
    register={[Function]}
    setValue={[Function]}
  />
  <FieldSelect
    errors={Object {}}
    fieldProps={Object {}}
    labelProps={Object {}}
    onChange={null}
    register={[Function]}
    setValue={[Function]}
  />
  <Button
    ariaLabel=""
    className=""
    disabled={false}
    elementLocator=""
    inheritedStyles=""
    primary={true}
    secondary={false}
    tertiary={false}
    transparent={false}
    type="submit"
    variation=""
  >
    Submit
  </Button>
</form>
`;
